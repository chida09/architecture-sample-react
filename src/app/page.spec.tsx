import '@testing-library/jest-dom'
import Page from './page'
import { render, screen } from '@testing-library/react'

describe('Page', () => {
  it('renders a heading', () => {
    const { asFragment } = render(<Page />)

    // h1タグ
    const heading1 = screen.getByRole('heading', { level: 1 })
    const text = screen.getByText('Docs')

    // h2タグが複数存在する場合
    const heading2 = screen.getByRole('heading', { name: 'Learn' })

    const listElements = screen.getAllByRole('listitem')

    // heading tag
    expect(heading1).toBeInTheDocument()
    expect(text).toBeInTheDocument()
    expect(heading2).toBeInTheDocument()
    // list tag
    expect(listElements).toHaveLength(3)
    // snapshot test
    expect(asFragment()).toMatchSnapshot()
  })
})
