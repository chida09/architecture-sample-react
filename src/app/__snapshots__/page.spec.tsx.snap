// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders a heading 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <div
      class="description"
    >
      <p>
        Get started by editing 
        <code
          class="code"
        >
          src/app/page.tsx
        </code>
      </p>
      <div>
        <a
          href="https://vercel.com?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          By 
          <img
            alt="Vercel Logo"
            class="vercelLogo"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="24"
            src="/vercel.svg"
            style="color: transparent;"
            width="100"
          />
        </a>
      </div>
    </div>
    <div
      class="center"
    >
      <img
        alt="Next.js Logo"
        class="logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="37"
        src="/next.svg"
        style="color: transparent;"
        width="180"
      />
    </div>
    <div
      class="grid"
    >
      <a
        class="card"
        href="https://nextjs.org/docs?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h1>
          Docs 
          <span>
            -&gt;
          </span>
        </h1>
        <p>
          Find in-depth information about Next.js features and API.
        </p>
      </a>
      <a
        class="card"
        href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Learn
        </h2>
        <p>
          Learn about Next.js in an interactive course with quizzes!
        </p>
      </a>
      <a
        class="card"
        href="https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Templates 
          <span>
            -&gt;
          </span>
        </h2>
        <p>
          Explore the Next.js 13 playground.
        </p>
      </a>
      <a
        class="card"
        href="https://vercel.com/new?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Deploy 
          <span>
            -&gt;
          </span>
        </h2>
        <p>
          Instantly deploy your Next.js site to a shareable URL with Vercel.
        </p>
      </a>
    </div>
    <ul>
      <li>
        test 1
      </li>
      <li>
        test 2
      </li>
      <li>
        test 3
      </li>
    </ul>
  </main>
</DocumentFragment>
`;
